### PR TITLE
Design updates and duplicate names fixes

### PR DESCRIPTION
**Pull request by abdullahcalisir12**

### CODE REVIEW

**Comment by VaughnVernon** [comment]
@Florian-Schoenherr Please review.

File: src/main/frontend/src/components/aggregates/AggregateDialog.svelte [hunk 1]
```diff
@@ -89,12 +89,17 @@
 		}
 	}
 
+	const isAggregateNameUnique = (value) => {
+		const result = $aggregateSettings.filter(aggregate => aggregate.aggregateName === value);
```
Comment (Florian-Schoenherr) [inline src/main/frontend/src/components/aggregates/AggregateDialog.svelte:93]: `.find` would be enough, I guess. Unique could be refactored out on all these, if you want to (function giving back a function).

File: src/main/frontend/src/components/aggregates/StateFields.svelte [hunk 2]
```diff
@@ -9,14 +9,25 @@
...
+  const deleteStateField = (index) => { stateFields.splice(index, 1); stateFields = stateFields; }
+  
+  const isFieldNameUnique = (value) => {
+    const result = stateFields.filter(field => field.name === value);
+		return result.length === 1 ? undefined : 'Name must be unique';
+  }
+
+  $: if (stateFields) {
+    stateFields.forEach(field => {
+      isFieldNameUnique(field.name);
```
Comment (Florian-Schoenherr) [inline src/main/frontend/src/components/aggregates/StateFields.svelte:21]: What are you trying to do here? The return value is not used.

**Comment by Florian-Schoenherr** [comment]
See review^
The other changes are all styling related, so if it looks good we can merge.
I will then take a look if I find any bugs or design issues.

File: src/main/frontend/src/components/aggregates/AggregateDialog.svelte (hunk 1 above)
Comment (abdullahcalisir12) [reply src/main/frontend/src/components/aggregates/AggregateDialog.svelte:93]: `.find` returns the first match, but I need to check if the value occurs more than one.
I tried closure(function return function) but it breaks error handling.

File: src/main/frontend/src/components/aggregates/StateFields.svelte (hunk 2 above)
Comment (abdullahcalisir12) [reply src/main/frontend/src/components/aggregates/StateFields.svelte:21]: I will update this

**Comment by abdullahcalisir12** [comment]
@Florian-Schoenherr thanks for the review, I will update PR according to your reviews

File: src/main/frontend/src/components/aggregates/AggregateDialog.svelte (hunk 1 above)
Comment (abdullahcalisir12) [reply src/main/frontend/src/components/aggregates/AggregateDialog.svelte]: `.find` return the first match, I want to show the error message if the value occurs more than 1. If I use `.find` I cannot be sure it's used multiple times or once

Comment (abdullahcalisir12) [reply src/main/frontend/src/components/aggregates/AggregateDialog.svelte]: fixed function

File: src/main/frontend/src/components/aggregates/StateFields.svelte (hunk 2 above)
Comment (abdullahcalisir12) [reply src/main/frontend/src/components/aggregates/StateFields.svelte]: I solved this via using keyed each block

**Comment by abdullahcalisir12** [comment]
@Florian-Schoenherr PR is ready to be merged if there are no design issues or any bugs you found related to this PR

**Comment by Florian-Schoenherr** [comment]
Looks nice! I will have a look at the Dialog (wanted to avoid cloning, but I should also look at the design).
Btw, why are you force-pushing? (it re-arranges the commits here in the PR, just saying 👍)

**Comment by abdullahcalisir12** [comment]
Great! I forget to push commits using --signoff, so to pass DCO check, I re-push with commits with --signoff

**Comment by Florian-Schoenherr** [comment]
Aha! Don't know what IDE you're using, on vscode there's a setting to auto-signoff (even with gpg).

Got this error:
image
Maybe on other pages too. You just forgot changing it in the `valid`-assignment.

**Comment by VaughnVernon** [comment]
@Florian-Schoenherr I looked at screen shots last night and thought they are good. Please confirm and let me know when to merge, or you can merge when ready.

**Comment by Florian-Schoenherr** [comment]
Small change: please make the AggregateDialog persistent (there's a prop to do that).

The bug is still not quite handled. The change on `formatArrayForSelect` was a nice idea, but that's only useful for making the Select not show the non-unique property. The data behind it is still wrong.
To fix this, on `AggregateDialog` there are some functions like `validField`, they need `isPropertyUnique`.
image

**Comment by abdullahcalisir12** [comment]
@Florian-Schoenherr what do you mean by making AggregateDialog persistent? And what's the prop?

**Comment by Florian-Schoenherr** [comment]
image
I came to that conclusion because I misclicked at one point (clicked outside of the Dialog) and this erases all changes (which is intended, but it should only close on button click)

**Comment by abdullahcalisir12** [comment]
@Florian-Schoenherr PR is updated!

**Comment by VaughnVernon** [comment]
@abdullahcalisir12 @Florian-Schoenherr I think I saw an indication of using localStorage. I would like to save the JSON to a local file and be able to reload it for edits. Perhaps the JSON could be saved along with project generation as well (as a backup). This is not urgent, but I think that losing people's work is a great way to make them hate your product.

**Review by Florian-Schoenherr — APPROVED** [review]

**Pull request merged**